backend/controller/timeline/query.go

```diff
@@ -205,6 +217,23 @@ func (s *Service) QueryTimeline(ctx context.Context, limit int, filters ...Timel
 		q += ")\n"
 	}
 
+	if len(filter.module) > 0 {
+		q += " AND ("
+		for i, module := range filter.module {
+			if i > 0 {
+				q += " OR "
+			}
+			if verb, ok := module.verb.Get(); ok {
+				q += fmt.Sprintf("((e.type = 'call' AND e.custom_key_3 = $%d AND e.custom_key_4 = $%d) OR (e.type = 'ingress' AND e.custom_key_1 = $%d AND e.custom_key_2 = $%d))",
```

Comment:
All verbs are handled by `e.type = 'call'` here. Eventually, this will have other `e.type`s as well, like maybe `cron`, `pubsub`, etc. Basically any event that can have a `module/verb` linked to it.

Reply:
Ahhhh got it, totally missed that on my first pass